server/src/main/java/org/opensearch/index/engine/EngineConfig.java

```diff
@@ -142,6 +143,13 @@ public Supplier<RetentionLeases> retentionLeasesSupplier() {
                 return s;
         }
     }, Property.IndexScope, Property.NodeScope);
+    public static final Setting<Integer> INDEX_CODEC_COMPRESSION_LEVEL_SETTING = Setting.intSetting(
```

Comment:
Can you add a comment explaining the levels, and which codecs support it?

Reply:
done